examples/jax/encoder/test_multiprocessing_encoder.py

```diff
@@ -0,0 +1,495 @@
+# Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# See LICENSE for license information.
+"""Encoder training with multi-GPU, multiprocessing and tesnor parallelism"""
+import argparse
+import unittest
+from functools import partial
+
+import jax
+import jax.numpy as jnp
+import nltk
+import numpy as np
+import optax
+import tensorflow_datasets as tfds
+from cuda import cudart
+from flax import linen as nn
+from flax.core.frozen_dict import FrozenDict
+from flax.linen import partitioning as nn_partitioning
+from flax.training import train_state
+from jax.experimental import mesh_utils
+from jax.experimental.pjit import pjit
+
+import transformer_engine.jax as te
+
+DEVICE_DP_AXIS = 'data'
+DEVICE_TP_AXIS = 'model'
+NAMED_BROADCAST_AXIS = 'my_broadcast_axis'
+NAMED_TP_AXIS = 'my_tp_axis'
+PARAMS_KEY = 'params'
+PARAMS_AXES_KEY = PARAMS_KEY + '_axes'
+DROPOUT_KEY = 'dropout'
+INPUT_KEY = 'input_rng'
+
+
+def check_num_gpu(desired_num_gpu):
+    """Check if the number of GPUs are correct."""
+    actual_num_gpu = len(jax.local_devices())
+    assert actual_num_gpu == desired_num_gpu, f"Number of GPUs is mismatch. " \
+        f"{desired_num_gpu} GPUs are assigned, but the actual number of GPUs is {actual_num_gpu}"
+
+
+def gpu_has_fp8():
```

Comment:
Looks like a duplicate from the `utils.py` and `fp8.py` functions

Reply:
Let's use those instead?

Reply:
Yes, is_fp8_available()` can be used instead. Although we don't expose is_fp8_available(), I think it should be fine for unit tests to use it, such as
```python
from transformer_engine.jax.utils import is_fp8_available
ret, reason = is_fp8_available()

@unittest.skipIf(not ret, reason=reason )
def test_te_fp8(self):
    """Test Transformer Engine with FP8"""
```

Reply:
50d9e4afd2787f45caf22d26eb13fd66ba58fbc3 solved this issue